docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -489,29 +489,40 @@ This could be useful to simplify the template structure:
 <1> The `name` will be resolved against the `item.parent`.
 <2> The `description` will be also resolved against the `item.parent`.
 
-It's also possible to specify an alias for the context object:
+This section might also come in handy when we'd like to avoid multiple expensive invocations:
 
 [source,html]
 ----
-{#with item.parent as myParent} <1>
-  <h1>{myParent.name}</h1>
+{#with item.callExpensiveLogicToGetTheValue(1,'foo',bazinga)}
+  {#if this is "fun"} <1>
+    <h1>Yay!</h1>
+  {#else}
+    <h1>{this} is not fun at all!</h1>
+  {/if}
 {/with}
 ----
-<1> `myParent` is the alias that can be used inside the tag.
+<1> `this` is the result of `item.callExpensiveLogicToGetTheValue(1,'foo',bazinga)`. The method is only invoked once even though the result may be used in multiple expressions.
 
-This section might also come in handy when we'd like to avoid multiple expensive invocations:
+===== Let/Set Section
 
+This section is an alternative to `{#with}` which allows to specify one or more named parameters:
 [source,html]
 ----
-{#with item.callExpensiveLogicToGetTheValue(true) as it}
-  {#if it is "fun"} <1>
-    <h1>Yay!</h1>
-  {#else}
-    <h1>{it} is not fun at all!</h1>
-  {/if}
-{/with}
+{#let myParent=order.item.parent}
+  <h1>{myParent.name}</h1>
+  <p>Price: {price}
```

Comment:
This `price` thing confuses me: it's not coming from `order.item.parent/myParent`, right?

Reply:
Nope. It's a mistake. Good catch! The original version had a `price` param too. Normally, Qute would attempt to find a `price` in the parent contexts...

Reply:
Well, it's fine that it would resolve it from the parent scope (that's what I expect), I just wanted to be sure, because its presence in this example confused me.